File: 백준/Gold/9084. 동전/동전.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()
	var tc, n, m int
	fmt.Fscanf(reader, "%d\n", &tc)
	for i := 0; i < tc; i++ {
		fmt.Fscanf(reader, "%d\n", &n)
		var coins = make([]int, n)
		for j := 0; j < n; j++ {
			fmt.Fscanf(reader, "%d ", &coins[j])
		}
		fmt.Fscanf(reader, "%d\n", &m)
		var dp = make([]int, m+1)
		dp[0] = 1
		for _, v := range coins {
			for j := 0; j <= m; j++ {
				if j >= v {
					dp[j] += dp[j-v]
				}
			}
		}
		fmt.Fprintf(writer, "%d\n", dp[m])
	}
}

File: 백준/Gold/9084. 동전/동전.py
import sys
input = sys.stdin.readline
sys.setrecursionlimit(10**6)
tc = int(input())
for _ in range(tc):
    n = int(input())
    coins = list(map(int, input().split()))
    m = int(input())
    dp = [0 for _ in range(m+1)]
    dp[0] = 1
    for coin in coins:
        for i in range(m + 1):
            if i >= coin:
                dp[i] += dp[i - coin]
    print(dp[m])